src/ui/modules/urlExport.ts

```diff
@@ -82,9 +81,42 @@ const urlExport = (parent, exportSettings: urlExportSettings, requestBody: urlEx
   } else {
     body = JSON.stringify(requestBody, null, 2)
   }
+  return body
+}
+
+const urlExport = (parent, exportSettings: urlExportSettings, requestBody: urlExportRequestBody) => {
+  // abort on missing url
+  if (exportSettings.url === '') {
+    // @ts-ignore
+    parent.postMessage({
+      pluginMessage: {
+        command: commands.closePlugin,
+        payload: {
+          notification: '🚨 No server url was provided, push aborted!'
+        }
+      } as PluginMessage
+    }, '*')
+  }
+  // init request
+  const request = new XMLHttpRequest()
+  // send to user defined url
+  request.open('POST', exportSettings.url)
+
+  addUrlExportRequestHeaders(request, exportSettings)
+
+  addUrlExportRequestEvents(request)
+
+  const body = generateUrlExportRequestBody(exportSettings, requestBody)
 
   // send request
   request.send(body)
 }
 
-export { urlExport }
+export {
+  urlExport,
+  addUrlExportRequestHeaders,
```

Comment:
Are you exporting everything apart from the urlExport for the sole purpose of testing?

If so, I think we should nest them all in a `_testing` object and export that. This cleans up the api a bit.

```js
_testing = {
  addUrlExportRequestHeaders: addUrlExportRequestHeaders,
  ...
}

export {
urlExport,
_testing
}

Reply:
@lukasoppermann I've made changes to use this pattern however I'm not familiar with many of the nuances here. Can you please review the changes I made to make sure it looks good. Unit tests succeed and my colleague was able to run this without problems.

Reply:
Hey, thank you very much, that sounds great. I am not sure this is the latest pattern, but it "soft-hides" internal exports and clearly marks them as testing only.

I'll have a look and merge this. Thank you for all the great work so far. 🙏